product_harmonized_system/models/product_product.py

```diff
@@ -0,0 +1,21 @@
+# Copyright 2011-2022 Akretion (http://www.akretion.com)
+# Copyright 2009-2022 Noviat (http://www.noviat.com)
+# @author Alexis de Lattre <alexis.delattre@akretion.com>
+# @author Luc de Meyer <info@noviat.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    def get_hs_code_recursively(self):
```

Comment:
This method should be private `_get_hs_code_recursively`
This is a good opportunity to also make private the other similar methods with the same name.

Public model methods are exposed to xmlrpc, it's a good practice to do that only when that's intended.

Reply:
In this case, it is nice to have that kind of method accessible via xmlrpc, isn't it ?

Reply:
But in such case it should return a json instead, or use the decorator `@api.return`.

Otherwise, the response would be something like this:

 ```json
{"result": "hs.code(123,)"}
```

As it is right now, it's better to have it private.

I wuold also argue that, if the idea is to expose this for external services.. it's better to move it to a computed field instead. That would save a roundtrip